ibm/mas_devops/roles/kafka/tasks/provider/redhat/check-amq-exists.yml

```diff
@@ -0,0 +1,40 @@
+---
```

Comment:
Think needs to be more generic and work both ways.

- If install strimzi check for redhat in the cluster
- If install redhat check for strimzi in the cluster

Use `operatorcondition` as the check ... e.g. ansible equivalent of:

```
oc get operatorcondition -l operators.coreos.com/amq-streams.amq-streams --all-namespaces
NAMESPACE     NAME                  AGE
amq-streams   amqstreams.v2.4.0-3   17h
```

And the equivalent for Strimzi.

I'm pretty sure there's example of this in the collection somewhere already too.

Reply:
That's good, thanks! I'll make these changes.
The only thing that concerns me is that both current and this approach uses namespace scoped lookups, so if users that changed the default namespaces while installing their kafka providers, we won't be able to determine if they're installed or not.

Reply:
In any case, i made the proposed changes so now it is generic enough to works both ways.

If installing strimzi and amq is already present in default namespace:
```
TASK [ibm.mas_devops.kafka : Assert: Stop Strimzi install if Red Hat AMQ Streams is already installed] ********************************************************************************************************************************************************
fatal: [localhost]: FAILED! => {
    "assertion": "not lookup_kafka_provider_exists",
    "changed": false,
    "evaluated_to": false,
    "msg": [
        "An existing Red Hat AMQ Streams installation was identified in this cluster under namespace: amq-streams",
        "Strimzi cannot be installed because it conflicts with Red Hat AMQ Streams service APIs. Please, uninstall Red Hat AMQ Streams before installing Strimzi.",
        "If uninstalling Red Hat AMQ Streams is not an option, then set 'kafka_provider: redhat' to continue using Red Hat AMQ Streams as this is the current installed Kafka provider."
    ]
}
```

If installing amq and strimzi is already present in default namespace:
```
TASK [ibm.mas_devops.kafka : Assert: Stop Red Hat AMQ Streams install if Strimzi is already installed] ********************************************************************************************************************************************************
fatal: [localhost]: FAILED! => {
    "assertion": "not lookup_kafka_provider_exists",
    "changed": false,
    "evaluated_to": false,
    "msg": [
        "An existing Strimzi installation was identified in this cluster under namespace: strimzi",
        "Red Hat AMQ Streams cannot be installed because it conflicts with Strimzi service APIs. Please, uninstall Strimzi before installing Red Hat AMQ Streams.",
        "If uninstalling Strimzi is not an option, then set 'kafka_provider: strimzi' to continue using Strimzi as this is the current installed Kafka provider."
    ]
}
```